src/Components/Admin/EnvironmentEditor/environment.js

```diff
@@ -102,16 +112,21 @@ export default function EnvironmentEditor({
         setScenes(items);
 
         const newSceneIds = items.map((scene) => scene.id);
-        const response = await editScenario({
-            id: environment.id,
-            name: environment.name,
-            friendly_name: environment.friendly_name,
-            description: environment.description,
-            scene_ids: newSceneIds,
-            is_published: environment.is_published,
-            is_previewable: environment.is_previewable,
-        });
-        setEnvironment(response.data);
+
+        try {
+            const response = await editScenario({
+                id: environment.id,
+                name: environment.name,
+                friendly_name: environment.friendly_name,
+                description: environment.description,
+                scene_ids: newSceneIds,
+                is_published: environment.is_published,
+                is_previewable: environment.is_previewable,
+            });
+            setEnvironment(response.data);
+        } catch (error) {
+            handleError(error);
+        }
```

Comment:
Do you think it's better to have the try catch here or should we handle it in the endpoints instead?

Reply:
I think it would actually make more sense to handle errors in the endpoints if possible and just pass the `handleError` function as well. This way, we don't need to remember to handle errors when implementing the frontend. Would reduce effort imo

Reply:
Yea I agree this would be cleaner, updated PR with this change.